frontend/Components/Header.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import {
+	Link,
+} from 'react-router-dom';
+export interface HeaderParams{
+	pageName: string;
+}
+export class MainHeader extends React.Component<HeaderParams> {
+	private readonly pageName: string;
+
+	constructor(params: HeaderParams) {
+		super(params);
+		this.pageName = params.pageName;
```

Comment:
In React you generally avoid having fields in component classes - you use [state](https://reactjs.org/docs/state-and-lifecycle.html) instead. The benefit of state is that any change to state will automatically trigger a rerender, wherein if you use fields you'll have to... actually there's probably no decent way to trigger a rerender manually.

Some pieces of wisdom:
1. In constructor you can set state simply by `this.state = {pageName: params.pageName}`
2. You read state simply by using `this.state`
2. Any subsequent changes to state have to be done via `this.setState()`.
3. `setState()` **IS ASYNCHRONOUS**!!!!! If you call `setState()` and on the next line try to read state it'll still have the old values.
4. Changes to state trigger rerenders of the component
5. If you set the state from props in constructor, when the props change the change won't propagate to state automatically again. React tries to keep the instances of the components alive. If for some reason you need to update the state if the props passed to a component change, you need [getDerivedStateFromProps](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops). But if you ever feel you need to ask me for feedback first before you go too deep into the rabbit hole.